presto-raptor/src/main/java/com/facebook/presto/raptor/storage/FileStorageService.java

```diff
@@ -29,20 +30,26 @@
 import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
+import java.util.concurrent.ExecutorService;
 import java.util.regex.Pattern;
 
 import static com.facebook.presto.raptor.RaptorErrorCode.RAPTOR_ERROR;
+import static io.airlift.concurrent.Threads.threadsNamed;
 import static java.util.Locale.ENGLISH;
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
 
 public class FileStorageService
         implements StorageService
 {
+    private static final Logger log = Logger.get(FileStorageService.class);
+
     private static final Pattern HEX_DIRECTORY = Pattern.compile("[0-9a-f]{2}");
     private static final String FILE_EXTENSION = ".orc";
 
     private final File baseStorageDir;
     private final File baseStagingDir;
+    private final ExecutorService executor = newSingleThreadExecutor(threadsNamed("background-staging-delete"));
```

Comment:
That assumption is fine since it's a private helper method. Though you could make it a no-args method named something like `deleteStagingFilesAsync` and put the listing logic in there.